### DISCUSSION TITLE
Side menu gives error of bad range when implementing with TCA

### DISCUSSION THREAD

**Discussion by holzerryan** [pointfreeco/swift-composable-architecture#3469] [General]
I'm attempting to implement a side menu or "hamburger menu", but I've run into an issue with the console outputting the following error : <decode: bad range for [%@] got [offs:394 len:1055 within:0]>
On the surface, everything appears to work correctly until the menu is presented and the view changes with the TabView.  My NoteListView works as expected with the main views and when you drill into detail views.  It stops working when the menu is used to change the selected tab for the TabView.

The tab view works correctly if you comment out the .toolbar code to hide the bottom tabbar.  drilling into subviews in the NotesListView do not work because it seems like you cannot interact with anything on the screen.  buttons and text fields don't seem to receive the tap gestures.

Any help would be appreciated.

My parent view code looks like this:

```
ZStack {
            
            TabView(selection: $store.selectedTab.sending(\.tabChanged)) {
                NavigationStack {
// This is an external library that is fully integrated with TCA.  We have a wrapper class that allows external apps to use the library without integrating with TCA.
                    InternalLibrary.NotesListView()
                    .navigationTitle("Notes")
                    .navigationBarTitleDisplayMode(.inline)
                    .toolbar {
                        ToolbarItem(placement: .topBarLeading) {
                            Button(action: {
                                store.send(.menuButtonTapped)
                            }) {
                                Text("Menu")
                            }
                        }
                    }
                }
                .tag(AuthenticatedFeature.Tab.notes)
                .toolbar(.hidden, for: .tabBar)
            }
            
            SideMenu(
                isShowing: $store.showMenu.sending(\.changeShowMenu),
                content: AnyView(
                    SideMenuView(
                        store: store.scope(
                            state: \.sideMenuFeature,
                            action: \.sideMenuFeature
                        )
                    )
                )
            )
        }
```

SideMenu is simple and looks like this:
```
struct SideMenu: View {
    
    @Binding var isShowing: Bool
    var content: AnyView
    var edgeTransition: AnyTransition = .move(edge: .leading)
    
    var body: some View {
        ZStack {
            if isShowing {
                Color.black
                    .opacity(0.3)
                    .ignoresSafeArea()
                    .onTapGesture {
                        isShowing.toggle()
                    }
                
                content
                    .transition(edgeTransition)
                    .background(
                        Color.clear
                    )
            }
        }
        .frame(maxWidth: .infinity, maxHeight: .infinity, alignment: .bottom)
        .ignoresSafeArea()
        .animation(.easeInOut, value: isShowing)
    }
}
```

**Comment by mbrandonw** [comment]
Hi @holzerryan, sorry it's hard to say without more info. Are you able to create a very small project that demonstrates the behavior you are seeing?

Also, fyi, the "<decode: bad range…" stuff is usually accompanied by purple runtime warnings in Xcode, which the library emits to notify users of incorrectly used APIs. Are you seeing any of those? What version of Xcode/iOS are you running?

**Reply by holzerryan** [reply]
I'll work on creating a small project that I can share...give me a little time and I'll let you poke around to see what I have missed.  I'm poking around now and it looks like the onAppear and onDisappear methods to not trigger for the swiftUI views when I'm selecting different menu items.  I'm not sure if it's a SwiftUI issue or a TCA issue at the moment.

**Comment by holzerryan** [comment]
I located the root cause to my problem.  The issue was that my InternalLibrary.NotesListView() function was generating a new store every time menu items were selected.  The quick/dirty fix was to create a variable at the top for a reference to a NoteListView so it's not recreated each time the menu selection is triggered.  The parent view held a reference to the noteListView.

var noteListView = InternalLibrary.NoteListView()

Then I simply used noteListView inside the TabView.

I'm intentionally not exposing the NoteListFeature to the outside world and I do not want to add public to all the classes and structures.  

My apologies for the red herring....the error I was seeing in the console didn't point me in the correct direction.  (happens from time to time)

**Reply by holzerryan** [reply]
Oh.  I'm using Xcode 16 on OS 15.0.  Probably doesn't matter in this situation, but I wanted to answer that question quick in case it solved a different problem.
